### ISSUE TITLE
Error when using max_nhcube=1

### ISSUE THREAD

**Issue by ndeutschmann** [gplepage/vegas#19]
Now that the option `max_nhcube=1` is taken into account thanks to v4.0.3, it seems to create a new issue as far as I understand.

Using the example from the VEGAS tutorial,

```python
import vegas
import math

def f(x):
    dx2 = 0
    for d in range(4):
        dx2 += (x[d] - 0.5) ** 2
    return math.exp(-dx2 * 100.) * 1013.2118364296088

integ = vegas.Integrator([[-1, 1], [0, 1], [0, 1], [0, 1]], max_nhcube=1)
```
I get the following error:

```python
IndexError                                Traceback (most recent call last)
<ipython-input-18-7a7d6e695d03> in <module>
     10 integ = vegas.Integrator([[-1, 1], [0, 1], [0, 1], [0, 1]], max_nhcube=1)
     11 
---> 12 result = integ(f, nitn=10, neval=1000)

_vegas.pyx in random_batch()

IndexError: Out of bounds on buffer access (axis 0)
```
Is this indeed a bug or am I misusing the option?

**Comment by gplepage** [comment]
Thanks for reporting the bug.

I see what is causing the problem and will fix it in the next release (next few days). Your fix (set ``max_nhcube`` in the call to ``integ()``) is correct and can be used until I fix the problem. A variation on your fix would be to omit ``max_nhcube=1`` when creating ``integ`` but include a line ``integ.set(max_nhcube=1)`` right after creating it.

**Comment by gplepage** [comment]
This issue is fixed in the version I just posted (v5.0).

**Comment by gplepage** [comment]
I am closing this issue because it is now fixed.
